templates/nextflow-ecs-service.yaml

```diff
@@ -0,0 +1,209 @@
+              - ec2:Describe*
+              - ec2:AuthorizeSecurityGroupIngress
+            Resource: '*'
+
+  EcsAlbSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupName: !Join
+        - '-'
+        - - !Ref 'AWS::StackName'
+          - EcsAlbSecurityGroup
+      GroupDescription: Security Group for Nextflow ECS load balancer
+      VpcId: !Ref VpcId
+
+  EcsAlbSecurityGroupIngress:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      GroupId: !GetAtt EcsAlbSecurityGroup.GroupId
+      IpProtocol: tcp
+      FromPort: !Ref AlbListenerPort
+      ToPort: !Ref AlbListenerPort
+      CidrIp: !Ref SecurityIngressCidrIp
+
+  EcsApplicationLoadBalancer:
+    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+    Properties:
+      Name: Nextflow-Tower-ALB
+      Scheme: internet-facing
+      LoadBalancerAttributes:
+      - Key: idle_timeout.timeout_seconds
+        Value: '30'
+      Subnets: !Ref SubnetIds
+      SecurityGroups:
+        - !GetAtt EcsAlbSecurityGroup.GroupId
+
+  EcsTargetGroup:
+    Type: AWS::ElasticLoadBalancingV2::TargetGroup
+    DependsOn: EcsApplicationLoadBalancer
+    Properties:
+      HealthCheckIntervalSeconds: 10
+      HealthCheckPath: /
+      HealthCheckProtocol: HTTP
+      HealthCheckTimeoutSeconds: 5
+      HealthyThresholdCount: 2
+      Port: !Ref TowerUIContainerPort
+      Protocol: HTTP
+      UnhealthyThresholdCount: 2
+      VpcId: !Ref VpcId
+
+  EcsAlbListener:
+    Type: AWS::ElasticLoadBalancingV2::Listener
+    DependsOn: EcsServiceRole
+    Properties:
+      DefaultActions:
+      - Type: forward
+        TargetGroupArn: !Ref EcsTargetGroup
+      LoadBalancerArn: !Ref EcsApplicationLoadBalancer
+      Certificates:
+        - CertificateArn: !Ref SSLCertificateArn
+      Port: !Ref AlbListenerPort
+      Protocol: HTTPS
+
+  TowerService:
+    Type: AWS::ECS::Service
+    DependsOn: EcsAlbListener
+    Properties:
+      Cluster: !Ref ClusterName
+      DesiredCount: 1
+      ServiceName: Nextflow-Tower-Service
+      TaskDefinition: !Ref TaskDefinitionArn
+      LaunchType: EC2
+      Role: !Ref EcsServiceRole
+      LoadBalancers:
+      - ContainerName: !Ref TowerUIContainerName
+        ContainerPort: !Ref TowerUIContainerPort
+        TargetGroupArn: !Ref EcsTargetGroup
+
+  AutoscalingRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service:
+              - application-autoscaling.amazonaws.com
+          Action:
+            - sts:AssumeRole
+      Path: /
+      Policies:
+      - PolicyName: service-autoscaling
+        PolicyDocument:
+          Statement:
+          - Effect: Allow
+            Action:
+              - application-autoscaling:*
+              - cloudwatch:DescribeAlarms
+              - cloudwatch:PutMetricAlarm
+              - ecs:DescribeServices
+              - ecs:UpdateService
```

Comment:
Out of curiosity, how do you get these specific lists of actions? Are they usually documented somewhere? Or do you pull these from examples?

Reply:
In this case I was working from an [AWS template snippet ](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/quickref-ecs.html)as a starting point.